src/editions/advanced-migrations.md

```diff
@@ -0,0 +1,173 @@
+# Advanced migration strategies
+
+## How migrations work
+
```

Comment:
Perhaps we can note again that migration works by migrates code to a state that compiles on both the current edition _and_ the next edition. This is why you migrate, then update the edition in Cargo.toml.

Reply:
Yea, sounds good. I wanted to make it clear that it is compatible on both editions, but maybe wasn't explicit enough. I added another paragraph trying to explain why it works that way.